.github/workflows/ci.yml

```diff
@@ -204,3 +203,26 @@ jobs:
           fail_ci_if_error: true
           token: ${{ secrets.CODECOV_TOKEN }}
           files: ./coverage.xml
+
+  Publish-Package:
```

Comment:
I am thinking when all will this job run. I don't have much expertise on GitHub actions but would it run on every PR getting merged to main looking at the `on` conditions in this yml?

Should we consider having a separate `deploy.yml` as the name for this file is `ci.yml `and also we might be able to separate the purpose well? I am looking at the cosmos repository's `deploy.yml` which also uses `hatch` I believe and they have the `on` event like this: https://github.com/astronomer/astronomer-cosmos/blob/main/.github/workflows/deploy.yml#L5

Reply:
Thanks, @pankajkoti , there were a couple of parts that I had commented on for testing it out (without the need for a release).
This action can only be triggered if it is a release: `if: github.event_name == 'release'`

I did consider separating - as we used to have - but I understand that for us to have something dependent upon (`needs`), it has to be in the same file. Cosmos releases things without checking if the tests are passing. The current approach is more reliable; what do you think? It is similar to what we adopted in the Python SDK (https://github.com/astronomer/astro-sdk/blob/main/.github/workflows/ci-python-sdk.yaml)

I agree the name CI is quite suboptimal. Do you have any suggestions?

Reply:
okay, understood.

I don't have a better suggestion actually. But we can also consider later build.yml, build_deploy.yml, or something like this. ci.yml is fine too at the moment.